src/visualization/plot_recipes.jl

```diff
@@ -284,20 +296,216 @@ end
 end
 
 
-# Create a PlotData2D plot directly from a TrixiODESolution for convenience
+
+"""
+    PlotData1D
+
+Holds all relevant data for creating 1D plots of multiple solution variables and to visualize the
+mesh.
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+struct PlotData1D{Coordinates, Data, VariableNames, Vertices} <:AbstractPlotData
+  x::Coordinates
+  data::Data
+  variable_names::VariableNames
+  mesh_vertices_x::Vertices
+end
+
+"""
+    PlotData1D(u, semi; solution_variables=cons2prim)
+
+Create a new `PlotData1D` object that can be used for visualizing 1D DGSEM solution data array
+`u` with `Plots.jl`. All relevant geometrical information is extracted from the semidiscretization
+`semi`. By default, the conservative variables from the solution are used for plotting.  This can be
+changed by passing an appropriate conversion function to `solution_variables`.
```

Comment:
Is this really true? As far as I can tell, always the conservative variables are used. If I am right, you should either remove this parameter, use hardcoded `cons2cons` for the solution variables, and add a `#TODO` comment that states that this will be fixed soon, or you need to convert the values of `u` by applying the `solution_variables` function to each node.

Reply:
Yes, I hardcoded it and added a comment with #TODO in ee3ece0

Reply:
On the coverage part I'm not sure on how to I can improve the else part and need help.

Reply:
First attempt to improve coverage is 0b1241f1. Let's wait and see if this is already sufficient.